### PR TITLE
Add a warning about signing messages without random nonce

### PR DESCRIPTION
**Pull request by KaffinPX**

### CODE REVIEW

**Comment by apoelstra** [comment]
Lol. This funcion definitely does **not** sign a message with a non-uniform nonce. You need to bend over backward using unsafe code to cause this library to produce a signature without a random nonce.

File: src/key.rs [hunk 1]
```diff
@@ -1016,6 +1016,8 @@ impl Keypair {
 
     /// Constructs a schnorr signature without aux rand for `msg` using the global
     /// [`SECP256K1`] context.
+    /// ⚠️ **Warning:** Signing multiple messages with the same private key without a unique nonce
+    /// can expose the private key. **This function signs the message without a random nonce!**
```
Comment (tcharding) [inline src/key.rs:1020]: nit: Needs a line of whitespace between the heading and body.

```suggestion
    ///
    /// ⚠️ **Warning:** Signing multiple messages with the same private key without a unique nonce
    /// can expose the private key. **This function signs the message without a random nonce!**
```

I'll leave review of the content to Andrew.

**Comment by KaffinPX** [comment]
> Lol. This funcion definitely does **not** sign a message with a non-uniform nonce. You need to bend over backward using unsafe code to cause this library to produce a signature without a random nonce.

Ah, didn't check its implementation, still less safe if compared w with_aux_rand right?

**Comment by apoelstra** [comment]
There is slightly less protection against hypothetical sidechannel attacks. We are not aware of any actual sidechannel attacks.

**Comment by apoelstra** [comment]
>Ah, didn't check its implementation

As politely as I can, why did you open a PR to change the doccomment of a function if you have no idea what it does and didn't even try to look?

**Comment by elichai** [comment]
The auxiliary randomness *is not* the nonce.

The auxiliary randomness exists only to mitigate specific kinds of power analysis side-channel attacks,
Providing it definitely improves security, but omitting it should not be considered dangerous, as most legacy signature schemes don't provide mitigations against such attacks.

Please do read BIP340: https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki

Specifically if you're interested in those side-channel attacks you can read more here: https://github.com/sipa/bips/issues/195


Maybe we should improve our docs a little bit regarding the auxiliary randomness

**Pull request closed without merging**